packages/ui/src/components/callout/callout.web.tsx

```diff
@@ -31,38 +31,71 @@ const calloutRecipe = cva({
       },
     },
   },
-  defaultVariants: {
-    variant: 'default',
-  },
 });
 
-type CalloutVariants = RecipeVariantProps<typeof calloutRecipe>;
+const icons = {
+  info: <InfoCircleIcon />,
+  success: <CheckmarkCircleIcon />,
+  warning: <ErrorTriangleIcon />,
+  error: <ErrorTriangleIcon />,
+  default: null,
+};
 
-interface CalloutProps extends BoxProps {
-  icon?: ReactNode;
+interface CalloutOwnProps {
+  /** Title of the callout.  */
   title?: string;
+  /** Contents of the callout */
+  children?: ReactNode;
+  /** Specify a custom icon to override the default for the variant. */
+  icon?: ReactNode;
+  /** Limit the content width to the specified value. Suitable for full-width callouts displayed at the top of the page. */
+  maxContentWidth?: SystemProperties['maxWidth'];
+  /** Render the callout in a portal. */
+  portal?: boolean;
+  /** Specify a custom element ID for portalled rendering.
+   * @default 'callout-portal' */
+  portalElementId?: string;
 }
-export function Callout(props: CalloutProps & CalloutVariants) {
-  const { children, icon, title, variant = 'default', ...rest } = props;
-  return (
-    <Box className={calloutRecipe({ variant })} data-variant={variant} {...rest}>
-      <Flag
-        align="top"
-        img={icon ? icon : getIconVariant(variant)}
+
+type CalloutVariants = RecipeVariantProps<typeof calloutRecipe>;
+
+export type CalloutProps = CalloutOwnProps & CalloutVariants & BoxProps;
+
+export function Callout({
+  children,
+  icon,
+  title,
+  variant = 'default',
+  maxContentWidth,
+  portal,
+  portalElementId = defaultPortalElementId,
+  ...rest
+}: CalloutProps) {
+  const content = (
+    <Box className={calloutRecipe({ variant })} role="status" {...rest}>
+      <Flex
+        justifyContent="space-between"
+        maxWidth={maxContentWidth}
+        mx="auto"
         p="space.05"
-        reverse
-        spacing="space.03"
+        gap="space.03"
+        color="ink.text-primary"
```

Comment:
Do we always need this color or should it be over-writable?

Reply:
This was carried over from before the refactor (just different location). We do need it to display the color correctly. Happens to over-writable too but incidentally.